core/sugartypes.ml

```diff
@@ -296,7 +292,17 @@ and cp_phrasenode =
   | CPLink        of Binder.with_pos * Binder.with_pos
   | CPComp        of Binder.with_pos * cp_phrase * cp_phrase
 and cp_phrase = cp_phrasenode WithPos.t
-                  [@@deriving show]
+and typename = (name * (quantifier * tyvar option) list * datatype' * Position.t)
+(* SJF: It would be nice to make these records at some point. *)
```

Comment:
I suggest making a ticket also. I very much like the idea. In fact I really missed the record syntax and the ability to nicely lay out comments next to the fields. Here's how things look in GHC source code:

```Haskell
  | RecordCon
      { rcon_con_name :: Located id         -- The constructor name;
                                            --  not used after type checking
      , rcon_con_like :: PostTc id ConLike  -- The data constructor or pattern synonym
      , rcon_con_expr :: PostTcExpr         -- Instantiated constructor function
      , rcon_flds     :: HsRecordBinds id } -- The fields
```
I think it would be easier to understand Links source code if data types had these kind of documentation.

Reply:
Thanks -- I've made a ticket: #533